source/filesystem.c

```diff
@@ -196,6 +196,17 @@ void addExecutableToMenu(menu_s* m, char* execPath)
 	addMenuEntryCopy(m, &tmpEntry);
 }
 
+void loadDescriptorNetloaderApp(descriptor_s* d, char* execPath) {
+	static char xmlPath[128];
+	snprintf(xmlPath, 128, "%s", execPath);
+	int l = strlen(xmlPath);
+	xmlPath[l-1] = 0;
+	xmlPath[l-2] = 'l';
+	xmlPath[l-3] = 'm';
+	xmlPath[l-4] = 'x';
```

Comment:
I don't know what setting the end of the string to be `xml` accomplishes. If anything, sounds like you need to go one more step and add the `.`.

There is also a problem if the length is too small; you can write to data before the buffer.